enhancements/installer/azurestack.md

```diff
@@ -0,0 +1,213 @@
+---
+title: azurestack
+authors:
+  - "@patrickdillon"
+reviewers:
+  - "@staebler"
+  - "@jhixson74"
+approvers:
+  - "@staebler"
+  - "@crawford"
+creation-date: 2021-03-10
+last-updated: 2021-03-10
+status: implementable
+see-also:
+  - "/enhancements/installer/azure-support-known-cloud-environments.md"
+---
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement covers adding support to install OpenShift clusters to Azure Stack Hub (ASH).
+Azure Stack Hub looks and feels like standard Azure, but differs significantly in terms of
+implemenetation and technical details. Azure Stack Hub shares characteristics from [Azure cloud
+environments](azure-support-known-cloud-environments.md) and on-prem platforms. This enhancement
+covers details for the Installer to create infrastructure and run an OpenShift cluster on ASH.
+
+## Motivation
+
+Currently, Azure Stack Hub is not supported but is a significant on-prem platform.
+
+### Goals
+
+1. Installer creates infrastructure and installs OpenShift on Azure Stack Hub.
+2. All standard support that entails, such as destroy.
+
+### Non-Goals
+
+1. ASH will not "inherit" established Azure
```

Comment:
It is easy enough to support these features. Since the installer is using the same platform for Azure Stack, the code already exists for support. The installer will use different terraform code, but that code is based largely on the existing terraform code for public Azure. It is easier to retain support than to try to remove support from the terraform code and add validation in the installer to prevent the user from using the feature. I agree that it should not be a **goal** to retain the full suite of feature, but we should not be explicitly disallowing features.